### DISCUSSION TITLE
Comfyui for full commercial use?

### DISCUSSION THREAD

**Discussion by MedleMedler** [Comfy-Org/ComfyUI#3804] [General]
Comfyui for full commercial use?

The unique strength of Comfyui is combining different (sub)models/data (software blocks) from third parties (with unprecedented results). But from a commercial point of view, this result in lots (and lots) of different legal user licenses.  

This means you have to give attributes to all third parties (sub-third parties, sub-sub_third parties (and so on)), which makes commercial use very limited. I do not think there is a way to use/sell pictures without the obligation to attribute, or is there?

Getting a commercial user license from Stabilty Ai (for their models), doesn't cover all the other (third party) user licenses. In theory you could commercially use Comfyui in combination with third parties who grant full commercial licenses. In practise this is almost impossible, because the most important third parties (e.g. IPAdapter, Controlnet) have open source (attribution) licenses. So what's the advantage of a commercial Stabilty Ai user license, if you want to use Comfyui fully commercial without attribution?

Conclusion: I do not think Comfyui can be used fully commercial (without attributing third parties) at this moment. As far as I know there is no overarching innitiative/organization that offers a fully closed (paid) Comfyui environment, by making deals with all important third parties. Or is there?

If my finding are correct and there is no overarching innitiative to make Comfyui suitable for fully commercial use, would you welcome such an initiative?

**Comment by ltdrdata** [comment]
I don't know about the others, but ipadapter's code and model are both under the Apache license, and ControlNet's code is under the Apache license while the model is under the OpenRAIL license, so I don't think it's much of a problem.

Of course, caution is needed with other technologies. For example, YOLO is licensed under AGPL.

**Reply by MedleMedler** [reply]
Thank you your answer. 

If I am correct, you have to give tribute to all the creators of all used models and custom nodes.  If you want to do it legally correct, you will also have to give attribute to the sub-sub models. E.g. if you use Controlnet, you have to give contribute to Controlnet. Then you also have to give attibute to the Auxiliry node and then you have to give attibute to all the pre-processors in the Auxiliri node you use. And I wouldn't be surprised that the preprocessors (like lineart) also are using third party software you have to give attribute to. This is a jungle of attributions.

This means (e.g.) that it is not suitable to sell a picture to photo stock agency. They have to resell the picture without the obligation for their customer to attribute. Another example: an advertising agency wants to use generated visuals for a new website of a big multinational who also do not want obligated attributions on their website.

So it depends on how (and for who) you want to use generated pictures. For personal use and for parties who don't mind multiple obligations to attribute,  I also see no issues :-)

For full commercial use (without obligations to contribute) I see issues (or am I missing something?).

**Reply by MedleMedler** [reply]
And thank you for mentioning this OpenRail license. There is no mention (and/or link) to this OpenRail license on the Huggingface page of Ipadapter at all. 

Which brings me to another weakness of Comfyui (every strengh has also it’s weaknesses) , seen from a full-commercial-use point of view. There are lots of contributers (hear, hear), varying from professionals to hobbyists. 
Because of the free character of Comfyui (which is of course wonderful), you cannot know if the creator can be trusted having all his licenses complete, mentioned, correct and up to date. You also cannot know if a third party creator has not used any other parts of sotware code that has license obligations. 

In my opinion a overarching organization is much needed, making license deals with third party creators (checking everything) and can guarantee a kind of closed source evironment for all the users who want to use the incredible power of a customizable Comfyui for full commercial use.
I even think this is essential for Comfyui to survive in future (a free and a paid version): open source Comfyui together with a (open) closed source Comfyui. 
In my opnion, this is win-win, because this will give third party developers an extra opportunity to make money (while still offering there software as free open source) and also makes professional use (in a full commercial way without obligations to contribute) fully possible.

**Reply by ltdrdata** [reply]
The IPAdapter model is not under openrail, but ControlNet is. For IPAdapter, it is listed as Apache on Huggingface.

https://huggingface.co/h94/IP-Adapter (apache-2.0)
https://huggingface.co/lllyasviel/ControlNet (openrail)

**Reply by ltdrdata** [reply]
There is a significant difference between distributing outputs created using code or models and modifying and redistributing the code or models themselves. 

If what you're trying to do is package ComfyUI to deliver the program to customers, you definitely need to include a list of licenses.
If there are no specific license restrictions on the model, there is no need to attach a license for the images you create.

**Reply by MedleMedler** [reply]
You are right, Openrail is for controlnet not for Ipadpter.

No I do not mean packaging Comfyui to deliver the program. I just mean pictures that are made with Comfyui and can be used without the obligation to give contribute to the software creators. 

So you are saying that these licenses are software licenses (and not end user licenses). Licenses, alter, rewrite Comfyui, Models and Custom nodes.
You say these licenses are not for giving legal rights for using Comfyui (to generate videos/pictures), licenses to use(sell) the pictures/videos that are made by Comfyui.

Makes sence actually (lol). I now know what you mean by the fact that OpenRail can be an "user" issue using Controlnet. At the same time Controlnet’s software license itself is not. Because using is not altering(rewriting) the software itself.

Thank you very much for clarifying. Very good news: no legal right end user issues at all (except Openrail) :-) 

And thank you (on behalf of the community) for all your nodes that you have written. For me it is a honor you answering my qeustions :-)

**Reply by ltdrdata** [reply]
I understand that in the case of openrail, there are no particular restrictions as long as it is not generating inappropriate content, such as illegal material.

**Comment by SaeidUnique** [comment]
Hey guys, I’m a production manager at an area rug company, we are looking for a better solution to using AI to generate high quality lifestyle commercial images. I heard about Comfyui but I'm not sure if this platform can figure out the perspective and turn our images into high quality lifestyle images in different angles and rooms.

**Reply by ltdrdata** [reply]
In that case, you may need to apply Depth ControlNet based on a 3D model.

You can start from here.
https://youtube.com/@archai3d
